docs/Users/Tutorials/External_Bears.rst

```diff
@@ -0,0 +1,528 @@
+static analysis routine in their favourite language.
+
+How Does This Work?
+-------------------
+
+By using the ``@external_bear_wrap`` decorator you will have all the necessary
+data sent to your external executable (filename, lines, settings) as a JSON
+string via ``stdin``. Afterwards, the analysis takes place in your executable
+that can be written in **literally** any language. In the end, you will have to
+provide the ``Results`` in a JSON string via stdout.
+
+In this tutorial, we will go through 2 examples where we create a very simple
+bear. The first example will use a **compiled** language, ``C++``, that creates
+a standalone binary whilst in the second example we will take a look at
+``JS`` that needs ``node`` in order to run out of the browser.
+
+External Bear Generation Tool
+-----------------------------
+
+If you really do not want to write any Python code, there is a tool
+`here <https://gitlab.com/coala/coala-bear-management>`__,
+``coala-bears-create``, that will create the wrapper for you. We will be using
+
+::
+
+    $ coala-bears-create -ext
+
+in order to generate the wrapper for the bear.
+
+Writing a Bear in C++
+---------------------
+
+The bear that will be created with this tutorial will check whether there is any
+**coala** spelled with a capital ``C`` since that is a horrible mistake for one
+to make.
+
+1. Create a new directory and make it your current working directory.
+2. Run ``coala-bears-create`` as mentioned above in order to create the wrapper
+   for our ``C++`` bear. Answer the first question with a path to your created
+   directory (since it should be the current one you can go with the default
+   value and just hit ``Enter``).
+3. The most important questions are the ones regarding the executable name and
+   the bear name. Use ``coalaCheckBear`` for the bear name and
+   ``coalaCheck_cpp`` for the executable name.
+4. The rest of the questions are not important (languages, developer name and
+   contact info, license, etc) to the tutorial and you can go with the defaults.
+   When you are prompted about ``settings`` answer ``no`` (default). After the
+   script is finished running there should be 2 files in your current directory:
+   ``coalaCheckBear.py`` (the wrapper) and ``coalaCheckBearTest.py``.
+5. This tutorial will not focus on testing so ignore the second file for now.
+   The wrapper should look similar to the code block presented below. Some code
+   has been cleaned for convenience of explanation.
+
+::
+
+    import os
+
+    from coalib.bearlib.abstractions.ExternalBearWrap import external_bear_wrap
+
+
+    @external_bear_wrap(executable='coalaCheck_cpp',
+                        settings={})
+    class coalaCheckBear:
+        """
+        Checks for coala written with uppercase 'C'
+        """
+        LANGUAGES = {'All'}
+        REQUIREMENTS = {'coalaCheck_cpp'}
+        AUTHORS = {'Me'}
+        AUTHORS_EMAILS = {'me@mail.com'}
+        LICENSE = 'GPLv3'
```

Comment:
Oh, no, that applies to the python code. The `C++` license should be in the `C++` file. I will swap it to AGPL to match the rest.